libraries: SPI: Hanlde multiple SPI instances

soburi opened the pull request:
Create multiple SPI instance if spis array contains plural elements. Declare these as 'SPI1', 'SPI2', ..., from second element of the array. (First element is already declare as 'SPI'.)

soburi committed:
libraries: SPI: handle multiple SPI instances

Create multiple SPI instances if the `spis` array contains plural elements.
Declare these as 'SPI1', 'SPI2', ..., from the second element of the array.
(The first element is already declared as 'SPI'.)

If `spis` is not defined but the DTS already defines `arduino-spi` and,
use it to define the first SPI instance.
The `arduion-spi` is usually defined with arduino_header in
boards definitions.

Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@gmail.com>

DhruvaG2000 commented on libraries/SPI/SPI.cpp:
```diff
@@ -121,5 +121,28 @@ void arduino::ZephyrSPI::begin() {}
+/* When PROP_LEN(spis) == 1, DT_FOREACH_PROP_ELEM work not correctly. */
 arduino::ZephyrSPI
     SPI(DEVICE_DT_GET(DT_PHANDLE_BY_IDX(DT_PATH(zephyr_user), spis, 0)));
+#endif // HAS_PORP(i2cs)
```
Pardon me, I didn't understand how I2C has any relevance here in SPI.cpp?

soburi replied:
Sorry, I verified the code but forgot to edit the comment. I'm using the structure that was originally used for I2C.

DhruvaG2000 commented on libraries/SPI/SPI.cpp:
```diff
@@ -121,5 +121,28 @@ void arduino::ZephyrSPI::begin() {}
 arduino::ZephyrSPI
     SPI(DEVICE_DT_GET(DT_PHANDLE_BY_IDX(DT_PATH(zephyr_user), spis, 0)));
+#endif // HAS_PORP(i2cs)
+/* If i2cs node is not defined, tries to use arduino_i2c */
```
`s/arduino_i2c/arduino_spi` ?

soburi replied:
Fixed it. Thanks.

soburi commented:
I'm going to put it back into draft because I have some concerns. Sorry.

DhruvaG2000 reviewed (commented):
@soburi  nothing against this PR, but can you add more in your commit about how this change helps us/ what feature it enables? Talk about what enhancement this is bringing in basically.

DhruvaG2000 commented:
Also, `s/Hanlde/handle` in $subject

Ayush1325 commented on libraries/SPI/SPI.cpp:
```diff
@@ -121,5 +121,28 @@ void arduino::ZephyrSPI::begin() {}
     SPI(DEVICE_DT_GET(DT_PHANDLE_BY_IDX(DT_PATH(zephyr_user), spis, 0)));
+#endif // HAS_PORP(spis)
+/* If spis node is not defined, tries to use arduino_spi */
+#elif DT_NODE_EXISTS(DT_NODELABEL(arduino_spi))
```
So where does the `arduino-spi` name come from? Are any boards already defining it using that label?

soburi replied:
It is usually define with `arduino_header` in `zephyr/boards` definitions.
This line is intended to take advantage of it if it is already defined.
And we can override with overlay in this package with defining `spis` property.

Ayush1325 reviewed (commented):
LGTM

soburi commented:
@DhruvaG2000 

> @soburi nothing against this PR, but can you add more in your commit about how this change helps us/ what feature it enables? Talk about what enhancement this is bringing in basically.

Added description to commit message. 

> Also, s/Hanlde/handle in $subject

Fixed it. Thank you.

DhruvaG2000 commented:
> > Also, s/Hanlde/handle in $subject
> 
> Fixed it. Thank you.

@soburi  Nope, I still see the wrong spelling in subject. https://github.com/zephyrproject-rtos/gsoc-2022-arduino-core/pull/113/commits

DhruvaG2000 commented on libraries/SPI/SPI.h:
```diff
@@ -56,7 +56,24 @@ class ZephyrSPI : public HardwareSPI {
+#define DECL_EXTERN_SPI_N(i)  extern arduino::ZephyrSPI SPI##i;
+#define DECLARE_EXTERN_SPI_N(n, p, i)                                                             \
+        COND_CODE_1(ARDUINO_SPI_DEFINED_##i, (DECL_EXTERN_SPI_0(i)), (DECL_EXTERN_SPI_N(i)))
+
```
Please fix above checkpatch warns @soburi .

>  WARNING: please, no spaces at the start of a line

DhruvaG2000 replied:
Was this fixed?
Can you please fix, rebase and force push if not done?
